service/src/main/java/org/cbioportal/service/impl/GenePanelServiceImpl.java

```diff
@@ -76,118 +80,215 @@ public GenePanel getGenePanel(String genePanelId) throws GenePanelNotFoundExcept
+        if (updatedSampleIds.size() < QUERY_LIMIT) {
+            genePanelDataForQueriedProfiles = genePanelRepository
+                .fetchGenePanelDataInMultipleMolecularProfiles(updatedMolecularProfileIds, updatedSampleIds);
+        } else {
+            genePanelDataForQueriedProfiles = genePanelRepository.fetchGenePanelDataInMultipleMolecularProfiles(new ArrayList<>(distinctMolecularProfileIds), null);
+        }
+
+        Map<Pair<String, String>, GenePanelData> genePanelDataSet = genePanelDataForQueriedProfiles
             .stream()
-            .collect(toMap(
-                    datum -> new Pair(datum.getMolecularProfileId(),
-                    datum.getSampleId()), Function.identity()));
-
-        return genePanelData.stream().map(datum -> {
-            String molecularProfileId = datum.getMolecularProfileId();
-            if(molecularProfileId.endsWith("_fusion")) {
-                String mutationProfileIdToSearch = molecularProfileId.replace("_fusion", "_mutations");
-                Pair key = new Pair(mutationProfileIdToSearch, datum.getSampleId());
-                if(genePanelDataSet.containsKey(key)) {
-                    GenePanelData mutationGenePanelData = genePanelDataSet.get(key);
-                    datum.setGenePanelId(mutationGenePanelData.getGenePanelId());
-                    datum.setProfiled(mutationGenePanelData.getProfiled());
-                }
+            .collect(toMap(datum -> new Pair(datum.getMolecularProfileId(), datum.getSampleId()), Function.identity()));
+
+        Map<String, MolecularProfile> molecularProfileMap = molecularProfileService
+            .getMolecularProfiles(new ArrayList<>(distinctMolecularProfileIds), "SUMMARY")
+            .stream()
+            .collect(toMap(MolecularProfile::getStableId, Function.identity()));
+
+        List<String> filteredStudyIds = new ArrayList<>();
+        List<String> filteredSampleIds = new ArrayList<>();
+        for (int i = 0; i < molecularProfileIds.size(); i++) {
+            String molecularProfileId = molecularProfileIds.get(i);
+            if (molecularProfileMap.containsKey(molecularProfileId)) {
+                filteredStudyIds.add(molecularProfileMap.get(molecularProfileId).getCancerStudyIdentifier());
+                filteredSampleIds.add(sampleIds.get(i));
             }
-            return datum;
-        }).collect(toList());
-        // TODO: remove this block after fusion are migrated to structural variant in database
-	}
+        }
 
-    @Override
-    public List<GenePanelData> fetchGenePanelDataInMultipleMolecularProfilesByPatientIds(
-            List<String> molecularProfileIds, List<String> patientIds) {
+        Map<Pair<String, String>, Sample> sampleSetWithSequencedInfo;
+
+        if (updatedSampleIds.size() < QUERY_LIMIT) {
+            sampleSetWithSequencedInfo = sampleService
+                .fetchSamples(filteredStudyIds, filteredSampleIds, "SEQUENCED")
```

Comment:
using "SEQUENCED" in the projection field seems contrary to expectations. The four members of the Projection enum are [ ID, SUMMARY, DETAILED, META ], so adding meaning to this argument in the service layer in order to process "SEQUENCED" differently worries me. I think it should be discussed at least.

Maybe adding a named argument to the service layer function will help make the meaning more clear .. such as 
public List<Sample> fetchSamples(
    List<String> studyIds,
    List<String> sampleIds,
    Boolean computeSampleSequencedAttributeByProfile,
    String projection)
This way it is kept clear which mode (Projection) the original api request from the user had, and the behavior sub-service layer interactions are requesting.
Also ... I think that all api endpoints which accept the Projection argument should behave correctly for all 4 possible projections supplied by the user. So I would test that this code functions correctly if the original API call is made with projection ID, META, SUMMARY, DETAILED {they should all show consistent sample counts}

Reply:
Not required anymore